docs/reporting.md

```diff
@@ -0,0 +1,122 @@
+behave in unpredictable ways. SRT tries to do its best to deal with the resulting problems, but like any other software of this kind, it isn't perfect. In many cases, "best effort" is all you can count on.
+
+That being said, it can always be made better. And so we warmly welcome everyone
+who can contribute improvements to SRT. 
+
+We encourage you to read the following guidelines, which are based on the experiences of previous contributors and are intended to make it easier for you to debug and report problems in a way that benefits the entire SRT community.
+
+
+Problem Reporting Guidelines
+============================
+
+1. We treat every problem report very seriously and will be doing our best to
+resolve them, but we need something to start the research with. When you report a problem, providing a description of
+the behavior and maybe error logs is a good start. But sometimes this isn't enough. If you can, try to replicate the
+behavior, and attach the debug log(s) and any pcap file(s) to your report.
+
+2. Sometimes problems result from a network that doesn't satisfy the
+minimum requirements for SRT. For example, the available bandwidth might not be enough to
+bear the traffic you are trying to send through it, or the latency might not be high
+enough to compensate for the network's maximum non-spiked RTT. We need to sort this kind of
+problem out first.
+
+3. A thorough description of your environment is very important. We will be trying to recreate it in our lab in order to be
+able to test your case ourselves. Note that in many situations this may not be possible. There may be some peculiarities in your environment or network configuration that you may not even be aware of. If you are using tc and netem for traffic
+shaping, there may be some distinct settings in the network that will make it impossible for us to see what you are seeing. This is another case where you can help us by providing debug logs and
+the pcap files.
+
+4. If you ever see the `IPE` (Internal Program Error) keyword in the error
+logs, please try to report that as a top priority (just check if it wasn't
+reported already). This reports the execution path that shall never be taken.
+
+
+5. *Do not hesitate to report any unexpected behavior*,
+even if you feel the information is incomplete. We have some tricks up our sleeves, as do other project members, that may help us fill in the blanks. And sometimes, Lady Luck is also on our side!
+
+
+
+
+Debug Logs
+==========
+
+The debug logs that can be generated with SRT provide very detailed descriptions of its internal behaviour. In
+fact, they can sometimes approach the equivalent of "record and
+replay" for a testing session. Having the debug logs collected is
+in most cases essential to start researching a potential problem.
+
+As SRT is very highly time-based software, however, the
+usability of a debugger is very limited. Debug logs put a great burden on the performance, and for this reason have been
```

Comment:
I have a feeling that the redacted text no longer reflects my original thought :) What I meant was that:

1. The debug logs actual purpose was to be close enough equivalent of "record and replay".
2. The fact, that the use of the debugger is very limited for such a time-based solution as SRT (or, maybe better said, time-dependent, or even more elaborate, the solution where timely execution is part of its correctness), is the exact reason why debug logs - a close approach to "record and replay" - are so essential.
3. The fact, that the debug logs put a great burden on the performance, is the reason why they are not enabled by default - which is a completely different matter to that why debug logs provide advantage over a debugger.